liquid/validation.go

```diff
@@ -46,17 +46,19 @@ func ValidateServiceInfo(srv ServiceInfo) error {
 }
 
 func validateServiceInfoImpl(srv ServiceInfo) (errs errorset.ErrorSet) {
-	for resName, resInfo := range srv.Resources {
-		if !resInfo.Topology.IsValid() {
-			errs.Addf(".Resources[%q] has invalid topology %q", resName, resInfo.Topology)
+	sortedResourceKeys := slices.Sorted(maps.Keys(srv.Resources))
+	for _, resName := range sortedResourceKeys {
```

Comment:
There is no need to introduce a variable here:
```suggestion
	for _, resName := range slices.Sorted(maps.Keys(srv.Resources)) {
```
Same in the loop below.